src/Data/Arrays.php

```diff
@@ -0,0 +1,439 @@
+
+    /**
+     * Add an element to an array using "dot" notation if it doesn't exist.
+     *
+     * @param array  $array Array to be evaluated
+     * @param string $key   Key
+     * @param string $opr   Notation like 'dot'
+     * @param 
+     *
+     * @since 3.0.0
+     *
+     * @return array
+     */
+    public static function add($array, $key, $value, $opr = null)
+    {
+        if (!self::has($array, $key, $opr)) {
+            self::set($array, $key, $value, $opr);
+        }
+
+        return $array;
+    }
+
+    /**
+     * Set an array item to a given value using "operator" notation.
+     *
+     * @param array  $array Array to be evaluated.
+     * @param string $key   Key
+     * @param mixed  $value Value
+     * @param string $opr   Notation like 'dot'
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+     */
+    public static function set(&$array, $key = null, $value = null, $opr = null)
+    {
+        if (null === $value) {
+            return $array;
+        }
+        if (null === $key) {
+            return $array = $value;
+        }
+        if (null === $opr) {
+            return $array[$key] = $value;
+        }
+
+        $keys = explode($opr, $key);
+        foreach ($keys as $ks => $key) {
+            if (!isset($array[$key])) {
+                $array = &$array[$key];
+            }
+        }
+        $array = $value;
+
+        return $array;
+    }
+
+    /**
+     * Get an item from an array using "operator" notation.
+     *
+     * @param array        $array Default array.
+     * @param array|string $keys  Keys to search.
+     * @param string       $opr   Operator notaiton.
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+     */
+    public static function get($array, $key = null, $default = null, $opr = null)
+    {
+        if (!self::isReallyArray($array)) {
+            return $default;
+        }
+        if (null === $key) {
+            return $array;
+        }
+        if (array_key_exists($key, $array)) {
+            return $array[$key];
+        }
+
+        if (null !== $opr) {
+
+            if (strpos($key, $opr) === false) {
+                return $array[$key] ?? $default;
+            }
+
+            foreach (explode($opr, $key) as $k) {
+                if (self::isReallyArray($array) && array_key_exists($k, $array)) {
+                    $array = $array[$k];
+                } else {
+                    return $default;
+                }
+            }
+
+            return $array;
+        }
+
+        return $default;
+    }
+
+    /**
+     * Determine if an item or items exist in an array using 'Operator' notation.
+     *
+     * @param array        $array Default array.
+     * @param array|string $keys  Keys to search.
+     * @param string       $opr   Operator notaiton.
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+     */
+    public static function has($array, $keys = null, $opr = null)
+    {
+        if (is_null($keys)) {
```

Comment:
It should use the `$keys === null` condition to check whether the `$keys` is null because of consistency.